examples/sysaudio/main.zig

```diff
@@ -40,12 +41,17 @@ pub fn update(app: *App, engine: *mach.Core) !void {
     while (engine.pollEvent()) |event| {
         switch (event) {
             .key_press => |ev| {
-                try app.device.start();
+                if (builtin.cpu.arch == .wasm32) try app.device.start();
```

Comment:
This isn't a wasm-specific API, it should be called here. What issue do you experience if it is?

Reply:
An assertion is triggered in libsoundio's alsa backend:
```
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

info: mach: found Vulkan backend on Integrated GPU adapter: Intel(R) Xe Graphics (TGL GT2), Intel open-source Mesa driver: Mesa 22.0.4

info: channels 2
info: sample_rate 48000

example-sysaudio: /home/desttinghim/Code/mach/libs/sysaudio/upstream/soundio/src/alsa.c:1438: int outstream_start_alsa(struct SoundIoPrivate *, struct SoundIoOutStreamPrivate *): Assertion `!osa->thread' failed.
The following command terminated unexpectedly:
cd /home/desttinghim/Code/mach && /home/desttinghim/Code/mach/zig-out/bin/example-sysaudio
error: the following build command failed with exit code 1:
/home/desttinghim/Code/mach/zig-cache/o/d0bdf9f6e450ee3ba9cbd1a840fb1b36/build /nix/store/z5mqbfapamm3rp7hbr8dcr5jf7kcxkmx-zig-0.10.0-dev.3900+ab4b26d8a/bin/zig /home/desttinghim/Code/mach /home/desttinghim/Code/mach/zig-cache /home/desttinghim/.cache/zig run-example-sysaudio
```

Reply:
If we have the soundio backend keep track of whether or not we're currently playing, perhaps that resolves that? I suggest adding an internal `playing: bool` field [here](https://github.com/hexops/mach/blob/main/libs/sysaudio/src/soundio.zig#L26-L30) and then [guarding `play` and `pause` here](https://github.com/hexops/mach/blob/f807c85232e6891ac6cde42bd2dff6e58efa5463/libs/sysaudio/src/soundio.zig#L113-L136)

Reply:
I tweaked your advice and made the field `started` instead. I need to double check that I'm using the API correctly now.